src/main/java/app/controller/graphicsEngine/GraphicsEngine.java

```diff
@@ -85,14 +87,27 @@ public Vector getIntersection(Ray r, ArrayList<Agent> agents, Agent currentAgent
         return intersection;
     }
 
-    private ArrayList<Boundary> collectBoundaries(Map map)
+    public ArrayList<Boundary> collectBoundaries(Map map)
     {
         ArrayList<Boundary> boundaries = new ArrayList<>();
         map.getFurniture()
                 .forEach(furniture -> boundaries.addAll(furniture.getBoundaries()));
         return boundaries;
     }
 
+    public ArrayList<Boundary> collectWallBoundaries(Map map)
+    {
+        ArrayList<Boundary> boundaries = new ArrayList<>();
+        for (Furniture f : map.getFurniture())
+        {
+            if (f.getType().name() == "WALL")
```

Comment:
Maybe adding "BORDER" types is a good idea?
the agent might try to move outside of the map otherwise, but not sure.

Reply:
We were thinking about it but we decided against it just for the ease of calculation, and it hasn't caused any problems so far. It works fine for the capture/evasion/target agents, but if exploration agents do need it we can change it later then.

Reply:
Adding "BORDER" type isn't possible because then it will set the whole graph to be an obstacle, as the BORDER is a rectangle of that size.